app/src/main/kotlin/at/bitfire/davdroid/sync/SyncNotificationManager.kt

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.sync
+
+import android.accounts.Account
+import android.app.PendingIntent
+import android.app.TaskStackBuilder
+import android.content.ContentUris
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.provider.CalendarContract
+import android.provider.ContactsContract
+import android.provider.Settings
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import at.bitfire.dav4jvm.exception.UnauthorizedException
+import at.bitfire.davdroid.R
+import at.bitfire.davdroid.db.Collection
+import at.bitfire.davdroid.resource.LocalCollection
+import at.bitfire.davdroid.resource.LocalContact
+import at.bitfire.davdroid.resource.LocalEvent
+import at.bitfire.davdroid.resource.LocalResource
+import at.bitfire.davdroid.resource.LocalTask
+import at.bitfire.davdroid.ui.DebugInfoActivity
+import at.bitfire.davdroid.ui.NotificationRegistry
+import at.bitfire.davdroid.ui.account.AccountSettingsActivity
+import at.bitfire.ical4android.TaskProvider
+import com.google.common.base.Ascii
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import dagger.hilt.android.qualifiers.ApplicationContext
+import okhttp3.HttpUrl
+import org.dmfs.tasks.contract.TaskContract
+import java.io.IOException
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class SyncNotificationManager @AssistedInject constructor(
+    @Assisted val account: Account,
+    @ApplicationContext private val context: Context,
+    private val logger: Logger,
+    private val notificationRegistry: NotificationRegistry
+) {
+
+    @AssistedFactory
+    interface Factory {
+        fun create(account: Account): SyncNotificationManager
+    }
+
+    /**
+     * Tries to inform the user that the content provider is missing or disabled.
+     */
+    fun notifyContentProviderError(authority: String) {
+        val (titleResource, textResource) = when {
+            authority == ContactsContract.AUTHORITY -> Pair(
+                R.string.sync_warning_contacts_storage_disabled_title,
+                R.string.sync_warning_contacts_storage_disabled_description
+            )
+            authority == CalendarContract.AUTHORITY -> Pair(
+                R.string.sync_warning_calendar_storage_disabled_title,
+                R.string.sync_warning_calendar_storage_disabled_description
+            )
+            TaskProvider.ProviderName.entries.map { it.authority }.contains(authority) -> Pair(
+                R.string.sync_warning_tasks_storage_disabled_title,
+                R.string.sync_warning_tasks_storage_disabled_description
+            )
```

Comment:
As you have mentioned previously, the tasks message should never appear. Now as we don't show the notification on `SecurityException` anymore, it really shouldn't appear. So we can take it out, together with the strings.

However if it's accidentally called with the task provider nevertheless, we have the `else branch`.